src/TesApi.Web/Management/Clients/TerraWsmApiClient.cs

```diff
@@ -68,13 +70,32 @@ public virtual async Task<ApiCreateBatchPoolResponse> CreateBatchPool(Guid works
         {
             ArgumentNullException.ThrowIfNull(apiCreateBatchPool);
 
-            var uri = GetCreateBatchPoolUrl(workspaceId);
+            HttpResponseMessage response = null;
+            try
+            {
+                var uri = GetCreateBatchPoolUrl(workspaceId);
+
+                Logger.LogInformation($"Creating a batch pool using WSM for workspace:{workspaceId}");
+
+                response =
+                    await HttpSendRequestWithRetryPolicyAsync(() => new HttpRequestMessage(HttpMethod.Post, uri) { Content = GetBatchPoolRequestContent(apiCreateBatchPool) },
+                        setAuthorizationHeader: true);
 
-            var response =
-                await HttpSendRequestWithRetryPolicyAsync(() => new HttpRequestMessage(HttpMethod.Post, uri) { Content = GetBatchPoolRequestContent(apiCreateBatchPool) },
-                    setAuthorizationHeader: true);
+                Logger.LogInformation($"Successfully created a batch pool using WSM for workspace:{workspaceId}");
 
-            return await GetApiResponseContentAsync<ApiCreateBatchPoolResponse>(response);
+                return await GetApiResponseContentAsync<ApiCreateBatchPoolResponse>(response);
+            }
+            catch (Exception ex)
+            {
+                var responseContent = "";
+                if (response is not null)
+                {
+                    responseContent = await response.Content.ReadAsStringAsync();
+                }
+
+                Logger.LogError($"Failed to create a batch pool using wsm. Response content:{responseContent}", ex);
```

Comment:
I thought that the exception goes before the message

Reply:
Thx. Will fix.